luarules/gadgets/unit_inherit_creation_xp.lua

```diff
@@ -0,0 +1,63 @@
+function gadget:GetInfo()
+	return {
+		name = "Inherit Creation Units XP",
+		desc = "Allows units with added UnitDefs to gain a defined fraction of the XP earned by their creations.",
+		author = "SethDGamre and Xehrath",
+		date = "May 2024",
+		license = "Public domain",
+		layer = 50,
+		enabled = true
+	}
+end
+
+-- synced only
+if not gadgetHandler:IsSyncedCode() then return false end
+
+local inheritchildrenxp = {} -- stores the value of XP rate to be derived from unitdef
+-- inheritxratemultiplier = defined in unitdef customparams of the parent unit. It's a number by which XP gained by children is multiplied and passed to the parent
+
+local childrenwithparents = {} --stores the parent/child relationships format. Each entry stores key of unitID with an array of {unitID, builderID, xpInheritance}
+for id, def in pairs(UnitDefs) do
+	if def.customParams then
+		if def.customParams.inheritxratemultiplier then
+			inheritchildrenxp[id] = def.customParams.inheritxratemultiplier or -1
```

Comment:
why the -1?

Reply:
i was copying something from the Workertime_boost gadget I wrote. I left as is for fear of breaking something. I think it was to avoid the value being nil.

Reply:
seems to work without it, so, I think you're implication is correct. Not necessary. Committed.